src/mds/CDir.cc

```diff
@@ -2974,6 +2994,14 @@ int CDir::_next_dentry_on_set(set<dentry_key_t>& dns, bool missing_okay,
     // okay, we got a  dentry
     dns.erase(dnkey);
 
+    if (dn->get_projected_version() < scrub_infop->last_recursive.version &&
+	!(scrub_infop->header && scrub_infop->header->force)) {
```

Comment:
Can we get here without having set the header? It's a required param of scrub_initialize so I think we're safe.